include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,388 @@
+    private:
+        template <typename... Sentinels>
+        constexpr sentinel(const Sentinels&... sentinels)
+            : end_(sentinels...) {}
+    public:
+        template <bool OtherConst>
+            requires (std::sentinel_for<std::conditional_t<!Const, std::ranges::sentinel_t<Views>,
+                                                                   std::ranges::sentinel_t<const Views>>,
+                                        std::conditional_t<!OtherConst, std::ranges::iterator_t<Views>,
+                                                                        std::ranges::iterator_t<const Views>>> && ...)
+        friend constexpr bool operator==(const iterator<OtherConst>& x, const sentinel& y)
+        {
+            return x.compare_with_sentinels(y.end_, std::make_index_sequence<sizeof...(Views)>());
+        }
+
+        template <bool OtherConst>
+            requires (std::sized_sentinel_for<std::conditional_t<!Const, std::ranges::sentinel_t<Views>,
+                                                                         std::ranges::sentinel_t<const Views>>,
+                                              std::conditional_t<!OtherConst, std::ranges::iterator_t<Views>,
+                                                                              std::ranges::iterator_t<const Views>>> && ...)
+        friend constexpr std::common_type_t<std::conditional_t<!Const, std::ranges::range_difference_t<Views>,
+                                                                       std::ranges::range_difference_t<const Views>>...>
+        operator-(const iterator<OtherConst>& x, const sentinel& y) {
+            return x.distance_to_sentinels(y.end_, std::make_index_sequence<sizeof...(Views)>());
+        }
+
+        template <bool OtherConst>
+            requires (std::sized_sentinel_for<std::conditional_t<!Const, std::ranges::sentinel_t<Views>,
+                                                                         std::ranges::sentinel_t<const Views>>,
+                                              std::conditional_t<!OtherConst, std::ranges::iterator_t<Views>,
+                                                                              std::ranges::iterator_t<const Views>>> && ...)
+        friend constexpr std::common_type_t<std::conditional_t<!Const, std::ranges::range_difference_t<Views>,
+                                                                       std::ranges::range_difference_t<const Views>>...>
+        operator-(const sentinel& y, const iterator<OtherConst>& x) {
+            return -(x - y);
+        }
+
+    private:
+        friend class zip_view;
+
+        using end_type = std::conditional_t<!Const, tuple_type<std::ranges::sentinel_t<Views>...>,
+                                                    tuple_type<std::ranges::sentinel_t<const Views>...>>;
+
+        end_type end_;
+    }; // class sentinel    
+
+    constexpr auto begin() requires (std::ranges::range<Views> && ...) // !simple_view?
```

Comment:
Please add a TODO comment or add a simple view constraint everywhere.

Reply:
It seems, we don;t need a kind of 'simple view' requirement... I 'll remove the comment.